pkg/cloud/services/networking/network.go

```diff
@@ -183,6 +230,29 @@ func (s *Service) ReconcileSubnet(clusterName string, openStackCluster *infrav1.
 	return nil
 }
 
+func (s *Service) getNetworkByID(networkID string) (networks.Network, error) {
+	opts := networks.ListOpts{
+		ID: networkID,
+	}
+
+	allPages, err := networks.List(s.client, opts).AllPages()
+	if err != nil {
+		return networks.Network{}, err
+	}
+
+	allNetworks, err := networks.ExtractNetworks(allPages)
+	if err != nil {
+		return networks.Network{}, err
+	}
+
+	switch len(allNetworks) {
+	case 0:
+		return networks.Network{}, nil
+	case 1:
+		return allNetworks[0], nil
+	}
+	return networks.Network{}, errors.New("too many resources")
```

Comment:
maybe we can consider update this error message

Reply:
Sure, updated to "multiple external network found"
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/573/files#diff-2bc6c3264c5dfec7bfdf49c2f3618635R255